SemanticUI/Modules/Dropdown.elm

```diff
@@ -0,0 +1,437 @@
+module SemanticUI.Modules.Dropdown exposing
+    ( Builder
+    , Config
+    , Dropdown(..)
+    , Variation(..)
+    , attributes
+    , disabled
+    , dropdown
+    , dropdownIcon
+    , fluid
+    , linkItem
+    , scrolling
+    , toCustomHtml
+    , toHtml
+    , toItem
+    , toggleEvent
+    )
+
+{-| Helper to create semantic-ui animated dropowns without the need for external JS.
+
+As an example a big dropdown menu using layout:
+
+    let
+        mainMenu { toDropdown, toToggle, drawer } =
+            toDropdown div
+                []
+                [ toToggle div [ class "text" ] [ text "File" ]
+                , drawer
+                    []
+                    [ Dropdown.toItem div [] [ text "New" ]
+                    , Dropdown.toItem div [] [ span [ class "description" ] [ text "ctrl + o" ], text "Open..." ]
+                    , Dropdown.toItem div [] [ span [ class "description" ] [ text "ctrl + s" ], text "Save as..." ]
+                    , Dropdown.toItem div [] [ span [ class "description" ] [ text "ctrl + r" ], text "Rename" ]
+                    , Dropdown.toItem div [] [ text "Make a copy" ]
+                    , Dropdown.toItem div [] [ i [ class "folder icon" ] [], text "Move to folder" ]
+                    , Dropdown.toItem div [] [ i [ class "trash icon" ] [], text "Move to trash" ]
+                    , div [ class "divider" ] []
+                    , Dropdown.toItem div [] [ text "Download As.." ]
+                    , Dropdown.dropdown
+                        { identifier = "fileSub"
+                        , onToggle = ToggleFileSub
+                        , state = model.fileSub
+                        }
+                        |> Dropdown.toggleEvent Toggle.OnHover
+                        |> Dropdown.dropdownIcon True
+                        |> Dropdown.toCustomHtml subMenu
+                    ]
+                ]
+
+        subMenu { toDropdown, toToggle, drawer } =
+            toDropdown (Dropdown.toItem div)
+                []
+                [ toToggle div [] [ text "Publish to Web" ]
+                , drawer
+                    []
+                    [ Dropdown.toItem div [] [ text "Google Docs" ]
+                    , Dropdown.toItem div [] [ text "Google Drive" ]
+                    , Dropdown.toItem div [] [ text "Google Drive" ]
+                    , Dropdown.toItem div [] [ text "Dropbox" ]
+                    , Dropdown.toItem div [] [ text "Adobe Creative Cloud" ]
+                    , Dropdown.toItem div [] [ text "Private FTP" ]
+                    , Dropdown.toItem div [] [ text "Another Service..." ]
+                    ]
+                ]
+    in
+    Dropdown.dropdown
+        { identifier = "file"
+        , onToggle = ToggleFile
+        , state = model.file
+        }
+        |> Dropdown.dropdownIcon True
+        |> Dropdown.toCustomHtml mainMenu
+
+-}
+
+import Dropdown
```

Comment:
I think this is not compiling at the moment am I correct ? Because there is no `Dropdown` module and no dependency exposing `Dropdown`, what is this `Dropdown` module and where is it coming from ?

Reply:
It's elm-dropdown from the original PR, except that it's currently using https://github.com/circuithub/elm-dropdown/pull/1 as mentioned in the related section.

I didn't spend a ton of time looking at elm-dropdown, do you think there's any problems in it as it stands?